native-tls-hook.js

```diff
@@ -56,6 +61,23 @@ TARGET_LIBS.forEach((targetLib) => {
     }
 });
 
+function hook_proxygen_SSLVerification(library) {
+    const functionName = "_ZN8proxygen15SSLVerification17verifyWithMetricsEbP17x509_store_ctx_stRKNSt6__ndk112basic_stringIcNS3_11char_traitsIcEENS3_9allocatorIcEEEEPNS0_31SSLFailureVerificationCallbacksEPNS0_31SSLSuccessVerificationCallbacksERKNS_15TimeUtilGenericINS3_6chrono12steady_clockEEERNS_10TraceEventE";
+
+    try {
+        const f = Module.getExportByName(library, functionName);
+
+        Interceptor.attach(f, {
+            onLeave: function (retvalue) {
+                retvalue.replace(1);
```

Comment:
This is definitely interesting, but it's tricky - right now all the existing scripts allow TLS interception just by the one extra MitM CA, but this line disables TLS validation _completely_ for this code :grimacing:

That creates some potential security problems. Even if you're intercepting Instagram traffic yourself, you probably don't want the risk that anybody on your local network could intercept you and steal your Instagram account.

All the other hooks handle this by using related APIs to actually validate the certificate, and just trust the one extra cert that's configured. That doesn't have this problem, but I think that's impractically difficult here since there's no public code & docs available for libliger anywhere I can see. We can't easily tell what certificate this function is handling.

I think there's probably a route through here. What do you think about adding a `ENABLE_INSECURE_HOOKS` variable in config.js, defaulting to false, and then:

* This hook checks that variable, inside this function here.
* If `false`, the first time the hook runs, it prints a "Insecure hook required" message with instructions: set `ENABLE_INSECURE_HOOKS=true` in config.js, but be aware that this is insecure and you could be intercepted by others on your network.
* If `true`, run this line as here and disable TLS.